http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/ResourcesServiceInterp.scala

```diff
@@ -0,0 +1,358 @@
+package org.broadinstitute.dsde.workbench.leonardo.http.service
+
+import cats.Parallel
+import cats.effect.Async
+import cats.implicits.toFunctorOps
+import cats.mtl.Ask
+import cats.syntax.all._
+import org.broadinstitute.dsde.workbench.google2.DiskName
+import org.broadinstitute.dsde.workbench.leonardo.{
+  AppContext,
+  AppName,
+  AppStatus,
+  CloudContext,
+  DiskStatus,
+  RuntimeName,
+  WorkspaceId
+}
+import org.broadinstitute.dsde.workbench.leonardo.db.{
+  appQuery,
+  clusterQuery,
+  kubernetesClusterQuery,
+  nodepoolQuery,
+  persistentDiskQuery,
+  DbReference,
+  DiskServiceDbQueries,
+  KubernetesServiceDbQueries
+}
+import org.broadinstitute.dsde.workbench.leonardo.http.{
+  ctxConversion,
+  CreateAppRequest,
+  CreateDiskRequest,
+  CreateRuntimeRequest,
+  CreateRuntimeResponse,
+  GetAppResponse,
+  GetPersistentDiskResponse,
+  GetRuntimeResponse,
+  ListAppResponse,
+  ListPersistentDiskResponse,
+  ListRuntimeResponse2,
+  UpdateDiskRequest,
+  UpdateRuntimeRequest
+}
+import org.broadinstitute.dsde.workbench.leonardo.model.{
+  LeoAuthProvider,
+  NonDeletableDisksInProjectFoundException,
+  NonDeletableRuntimesInProjectFoundException
+}
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.model.UserInfo
+
+import scala.concurrent.ExecutionContext
+
+final class ResourcesServiceInterp[F[_]: Parallel](authProvider: LeoAuthProvider[F])(implicit
+  F: Async[F],
+  dbRef: DbReference[F],
+  ec: ExecutionContext
+) extends ResourcesService[F]
+    with RuntimeService[F]
```

Comment:
I don't want to reinvent the wheel, and make use of the existing services that already solved the listRuntimes, listApps, and listDisks problem. 
I cannot make this class abstract because we instantiate it in `boot.scala`, but having to override a gazillion of functions at the end of this file is clearly not the solution either.

I would appreciate any scala best practice here :)

Reply:
I'm not sure I would reach for inheritance here. What if you used composition instead -- e.g. have `ResourcesServiceInterp` take `RuntimeService`, `AppService`, and `DiskService` as constructor-args. Then you could call `runtimeService.deleteRuntime()`, etc.